content/token22-metadata.md

```diff
@@ -0,0 +1,1083 @@
+
+The added fields and functions in the metadata extension follow the [Token-Metadata Interface](https://github.com/solana-labs/solana-program-library/tree/master/token-metadata/interface)
+
+When a mint is initialized with the metadata extension, it will store these extra fields:
+```rust
+type Pubkey = [u8; 32];
+type OptionalNonZeroPubkey = Pubkey; // if all zeroes, interpreted as `None`
+
+pub struct TokenMetadata {
+    /// The authority that can sign to update the metadata
+    pub update_authority: OptionalNonZeroPubkey,
+    /// The associated mint, used to counter spoofing to be sure that metadata
+    /// belongs to a particular mint
+    pub mint: Pubkey,
+    /// The longer name of the token
+    pub name: String,
+    /// The shortened symbol for the token
+    pub symbol: String,
+    /// The URI pointing to richer metadata
+    pub uri: String,
+    /// Any additional metadata about the token as key-value pairs. The program
+    /// must avoid storing the same key twice.
+    pub additional_metadata: Vec<(String, String)>,
+}
+```
+
+With these added fields, the `@solana/spl-token-metadata` library has been updated with the following functions to help out:
+- `createInitializeInstruction`
+- `createUpdateFieldInstruction`
+- `createRemoveKeyInstruction`
+- `createUpdateAuthorityInstruction`
+- `createEmitInstruction`
+- `pack`
+- `unpack`
+
+Additionally, the @solana/spl-token library introduces a new function and two constants:
+- `getTokenMetadata`
+- `LENGTH_SIZE`: a constant number of bytes of the length of the data
+- `TYPE_SIZE`: a constant number of bytes of the type of the data
+
+The function `createInitializeInstruction` initializes the metadata in the account and sets the primary metadata fields (name, symbol, URI). The function then returns an instruction that will set the metadata fields in the mint account.
+
+This function takes eight parameters:
+  - `mint`: the mint account that will be initialize
+  - `metadata`: the metadata account that will be created
+  - `mintAuthority`: the authority that can mint tokens
+  - `updateAuthority`: the authority that can sign to update the metadata
+  - `name`: the longer name of the token
+  - `symbol`: the shortened symbol for the token, also known as the ticker
+  - `uri`: the token URI pointing to richer metadata
+  - `programId`: the SPL Token program ID (in this case it will be the Token Extension program ID)
+
+```ts
+export interface InitializeInstructionArgs {
+    programId: PublicKey;
+    metadata: PublicKey;
+    updateAuthority: PublicKey;
+    mint: PublicKey;
+    mintAuthority: PublicKey;
+    name: string;
+    symbol: string;
+    uri: string;
+}
+
+export function createInitializeInstruction(args: InitializeInstructionArgs): TransactionInstruction {
+    ...
```

Comment:
`// TODO` ?

Reply:
No, its just to show the body of the function, which are not nesseccary to actually show. More clearly, we can just have the function declaration